lib/command/src/Obelisk/Command/Run.hs

```diff
@@ -88,21 +86,44 @@ run = do
       , "Frontend.frontend"
       ]
 
+-- | Nix syntax requires relative paths to be prefixed by @./@ or
+-- @../@. This will make a 'FilePath' can be embedded in a Nix
+-- expression.
+toNixPath :: FilePath -> FilePath
+toNixPath root | "/" `isInfixOf` root = root
+               | otherwise = "./" <> root
+
 runRepl :: MonadObelisk m => m ()
 runRepl = do
-  pkgs <- getLocalPkgs
+  pkgs <- withProjectRoot "." getLocalPkgs
   withGhciScript pkgs $ \dotGhciPath -> do
     runGhciRepl dotGhciPath
 
 runWatch :: MonadObelisk m => m ()
 runWatch = do
-  pkgs <- getLocalPkgs
+  pkgs <- withProjectRoot "." getLocalPkgs
   withGhciScript pkgs $ \dotGhciPath -> runGhcid dotGhciPath Nothing
 
--- | Relative paths to local packages of an obelisk project
--- TODO a way to query this
-getLocalPkgs :: Applicative f => f [FilePath]
-getLocalPkgs = pure ["backend", "common", "frontend"]
+-- | Relative paths to local packages of an obelisk project.
+--
+-- These are a combination of the obelisk predefined local packages,
+-- and any packages that the user has set with the @packages@ argument
+-- to the Nix @project@ function.
+getLocalPkgs :: (MonadObelisk m, MonadIO m) => FilePath -> m [FilePath]
+getLocalPkgs root = do
+  root' <- liftIO $ makeAbsolute root
+  projectPackages <- readProcessJSONAndLogStderr Debug $
+    proc "nix"
+      [ "eval"
+      , "(let proj = import " <> root' <> " {}; in (map toString (proj.obelisk.reflex-platform.nixpkgs.lib.attrValues proj.passthru.packages)))"
```

Comment:
What do you think of slapping an `or {}` after `proj.passthru.packages` just in case that attr path isn't valid? I'm not sure I want obelisk to completely stop working if this changes. However, I also don't like silent "failure" so I'm on the fence. Possibly even adding an explicit `or error "proj.passthru.packages is not defined in your default.nix but is needed by obelisk"`.

Reply:
I added the defensive `proj.passthru.packages or {}`. But actually it's always defined because of a default parameter value.